File: 1-Easy/ASCII Art/ascii-art.js
const ALPHABET = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ?'.split('');
const LENGTH = parseInt(readline());
const HEIGHT = parseInt(readline());
const TEXT = readline();
const letterIndexes = [];

// put the indexes of the letters into an array
for (const char of TEXT) {
    if (ALPHABET.includes(char.toUpperCase())) {
        // the index of character multiplied by LENGTH
        letterIndexes.push((ALPHABET.indexOf(char.toUpperCase()) + 1) * LENGTH);
    }
    else {
        // the index of the last element, ?, multiplied by LENGTH
        letterIndexes.push(ALPHABET.length * LENGTH);
    }
}

let asciiArt = "";
for (let i = 0; i < HEIGHT; i++) {
    const ROW = readline();
    for (const index of letterIndexes) {
      // slice ROW at the given start and end, then append that to the asciiArt string
      asciiArt += ROW.slice(index - LENGTH, index);
    }
    asciiArt += "\n";
}

console.log(asciiArt);

File: 1-Easy/ASCII Art/ascii-art.cs
using System;
using System.Linq;

class Solution
{
    static void Main(string[] args)
    {
        char[] ALPHABET = "ABCDEFGHIJKLMNOPQRSTUVWXYZ?".ToCharArray();
        int LENGTH = int.Parse(Console.ReadLine());
        int HEIGHT = int.Parse(Console.ReadLine());
        string TEXT = Console.ReadLine().ToUpper();
        int[] letterIndexes = new int[TEXT.Length];
        
        // put the indexes of the letters into an array
        for (int i = 0; i < TEXT.Length; i++)
        {
            if (ALPHABET.Contains(TEXT[i]))
            {
                // the index of character multiplied by LENGTH
                letterIndexes[i] = (Array.FindIndex(ALPHABET, c => c == TEXT[i]) + 1) * LENGTH;
            }
            else
            {
                // the index of the last element, ?, multiplied by LENGTH
                letterIndexes[i] = ALPHABET.Length * LENGTH;
            }
        }

        string asciiArt = "";
        for (int i = 0; i < HEIGHT; i++)
        {
            string ROW = Console.ReadLine();
            foreach (int index in letterIndexes)
            {
                // substring ROW at the given start and end, then append that to the asciiArt string
                asciiArt += ROW.Substring(index - LENGTH, LENGTH);
            }
            asciiArt += "\n";
        }

        Console.WriteLine(asciiArt);
    }
}